api/v1alpha1/clustertemplate_types.go

```diff
@@ -28,9 +28,24 @@ type ClusterTemplateSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// JSON formatted string that includes the schema for the ClusterTemplate.
-	// Required values should be explicitly defined.
-	InputDataSchema string `json:"inputDataSchema"`
+	Templates       Templates       `json:"templates"`
+	InputDataSchema InputDataSchema `json:"inputDataSchema"`
+}
+
+// Templates defines the references to the templates required for ClusterTemplate.
+type Templates struct {
+	// ClusterInstanceDefaults defines a reference to a configmap with default values for ClusterInstance
+	ClusterInstanceDefaults string `json:"clusterInstanceDefaults"`
```

Comment:
I think this should be required since we expect some defaults from the configmap and some from hw manager. In theory, ClusterRequest is sent from SMO but it wouldn't be able to know all the input values.